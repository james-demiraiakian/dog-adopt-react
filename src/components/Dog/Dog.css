.dog-detail {
  margin-top: 10vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100vw;
}

h4 {
  width: 50vw;
  text-align: center;
}
