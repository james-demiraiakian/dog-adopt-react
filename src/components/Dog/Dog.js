import React from 'react';
import { Link } from 'react-router-dom';
import './Dog.css';

export default function Dog({ id, age, bio, breed, name, image }) {
  return (
    <div className="dog-detail">
      <h2>{name}</h2>
      <img className="dog-detail-image" src={image} />
      <h4>Age: {age}</h4>
      <h4>Breed: {breed}</h4>
      <h4>Bio: {bio}</h4>
      <Link to={`/dogs/${id}/edit`}>Edit</Link>
    </div>
  );
}
