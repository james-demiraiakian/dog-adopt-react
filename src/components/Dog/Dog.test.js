import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Dog from './Dog';

test('<Dog /> should render doggo info', () => {
  const { container } = render(
    <MemoryRouter>
      <Dog age="100" bio="hfdjska fhjdk hfjdksl" breed="Doggo" name="Blarg" image="qwop" />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
