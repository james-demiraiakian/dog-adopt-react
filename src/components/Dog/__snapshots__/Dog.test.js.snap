// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dog /> should render doggo info 1`] = `
<div>
  <div
    class="dog-detail"
  >
    <h2>
      Blarg
    </h2>
    <img
      class="dog-detail-image"
      src="qwop"
    />
    <h4>
      Age: 
      100
    </h4>
    <h4>
      Breed: 
      Doggo
    </h4>
    <h4>
      Bio: 
      hfdjska fhjdk hfjdksl
    </h4>
    <a
      href="/dogs/undefined/edit"
    >
      Edit
    </a>
  </div>
</div>
`;
